### PR TITLE
Add config script

### LINKED ISSUE #51: PS script to add config
Hi,

based on [Jason Haley's](http://jasonhaley.com/post/Azure-Help-WebApps-Copying-and-Exporting-Connection-Strings-and-App-Settings) scripts I did create a PS script to easily add new web app's to the config:

    .\AddWebAppToLetsEncrypt.ps1 -webAppTarget "AppName" -webAppHosts "appname.com"

Here the script, if someone else wants to use it:

```powershell
param(
	[string]$webAppTarget,
	[string]$webAppHosts,
	[string]$renewXNumberOfDaysBeforeExpiration = "-1", # set this to e.g. 65 after first request to avoid hitting the cert-limit if the job is triggerd manually more than 5 times
	[string]$tenantId = "<default tenant Id>",
	[string]$subscriptionId = "<default subscription Id>",
	[string]$webAppLetsEncrypt = "<default letsencrypt web app>",
	[string]$resourceGroupLetsEncrypt = "<default resource group of letsencrypt web app>",
	[string]$clientId = "<default client Id>",
	[string]$clientSecret = "<default client secret>",
	[string]$resourceGroupTarget = "<default target resource group>",
	[string]$email = "<default email>")

Login-AzureRmAccount

Set-AzureRmContext -SubscriptionId $subscriptionId

# Load Existing Web App settings for source and target
$webAppSource = Get-AzureRmWebAppSlot -ResourceGroupName $resourceGroupLetsEncrypt -Name $webAppLetsEncrypt -Slot "production"

# Get reference to the source app settings
$appSettingsSource = $webAppSource.SiteConfig.AppSettings

# Create Hash variable for App Settings
$appSettingsTarget = @{}

# Copy over all Existing App Settings to the Hash
ForEach ($appSettingSource in $appSettingsSource) {
    $appSettingsTarget[$appSettingSource.Name] = $appSettingSource.Value
}

# Add new settings
$appSettingsTarget["letsencrypt:" + $webAppTarget + "-clientId"] = $clientId
$appSettingsTarget["letsencrypt:" + $webAppTarget + "-email"] = $email
$appSettingsTarget["letsencrypt:" + $webAppTarget + "-hosts"] = $webAppHosts
$appSettingsTarget["letsencrypt:" + $webAppTarget + "-renewXNumberOfDaysBeforeExpiration"] = $renewXNumberOfDaysBeforeExpiration
$appSettingsTarget["letsencrypt:" + $webAppTarget + "-resourceGroup"] = $resourceGroupTarget
$appSettingsTarget["letsencrypt:" + $webAppTarget + "-subscriptionId"] = $subscriptionId
$appSettingsTarget["letsencrypt:" + $webAppTarget + "-tenantId"] = $tenantId 
if ($appSettingsTarget.ContainsKey("letsencrypt:webApps")) {
	if (!$appSettingsTarget["letsencrypt:webApps"].ToLower().Contains($webAppTarget.ToLower())) {
		$appSettingsTarget["letsencrypt:webApps"] = $appSettingsTarget["letsencrypt:webApps"] + ";" + $webAppTarget
	}
}
else {
	$appSettingsTarget["letsencrypt:webApps"] = $webAppTarget
}

# Save Settings to Target
Set-AzureRmWebAppSlot -ResourceGroupName $resourceGroupTarget -Name $webAppLetsEncrypt -Slot "production" -AppSettings $appSettingsTarget

# Get reference to the source Connection Strings
$connectionStringsSource = $webAppSource.SiteConfig.ConnectionStrings

# Create Hash variable for Connection Strings
$connectionStringsTarget = @{}

# Copy over all Existing Connection Strings to the Hash
ForEach($connStringSource in $connectionStringsSource) {
    $connectionStringsTarget[$connStringSource.Name] = @{ Type = $connStringSource.Type.ToString(); Value = $connStringSource.ConnectionString }
}

# Add new Connection String
$connectionStringsTarget["letsencrypt:" + $webAppTarget + "-clientSecret"] = @{ Type = "Custom"; Value = $clientSecret }

# Save Connection Strings to Target
Set-AzureRmWebAppSlot -ResourceGroupName $resourceGroupTarget -Name $webAppLetsEncrypt -Slot "production" -ConnectionStrings $connectionStringsTarget
```

### PR DESCRIPTION
**Pull request by chrfin**
This fixes #51

### CODE REVIEW

File: WebAppConfig.ps1 [hunk 1]
```diff
@@ -0,0 +1,67 @@
+param(
```
Comment (ohadschn) [inline WebAppConfig.ps1]: Please add `$ErrorActionPreference = "Stop"` and `Set-StrictMode -Version Latest` to the top

File: WebAppConfig.ps1 [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	[string]$webAppTarget,
+	[string]$webAppHosts,
+	[string]$renewXNumberOfDaysBeforeExpiration = "-1", # set this to e.g. 65 after first request to avoid hitting the cert-limit if the job is triggerd manually more than 5 times
+	[string]$tenantId = "<default tenant Id>",
+	[string]$subscriptionId = "<default subscription Id>",
+	[string]$webAppLetsEncrypt = "<default letsencrypt web app>",
+	[string]$resourceGroupLetsEncrypt = "<default resource group of letsencrypt web app>",
+	[string]$clientId = "<default client Id>",
+	[string]$clientSecret = "<default client secret>",
```
Comment (ohadschn) [inline WebAppConfig.ps1]: This is probably not something you want to check in with the default value, I would think this should get passed in the command-line every time

File: WebAppConfig.ps1 [hunk 3]
```diff
@@ -0,0 +1,67 @@
+param(
+	[string]$webAppTarget,
```
Comment (ohadschn) [inline WebAppConfig.ps1]: PowerShell parameter names should be PascalCase

File: WebAppConfig.ps1 [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	[string]$clientSecret = "<default client secret>",
+	[string]$resourceGroupTarget = "<default target resource group>",
+	[string]$email = "<default email>")
+
+Login-AzureRmAccount
+
+Set-AzureRmContext -SubscriptionId $subscriptionId
+
+# Load Existing Web App settings for source and target
+$webAppSource = Get-AzureRmWebAppSlot -ResourceGroupName $resourceGroupLetsEncrypt -Name $webAppLetsEncrypt -Slot "production"
```
Comment (ohadschn) [inline WebAppConfig.ps1]: Perhaps the slot could be parameterized? (here and in `Set-AzureRmWebAppSlot` below)

File: WebAppConfig.ps1 [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+# Add new settings
+$appSettingsTarget["letsencrypt:" + $webAppTarget + "-clientId"] = $clientId
+$appSettingsTarget["letsencrypt:" + $webAppTarget + "-email"] = $email
+$appSettingsTarget["letsencrypt:" + $webAppTarget + "-hosts"] = $webAppHosts
+$appSettingsTarget["letsencrypt:" + $webAppTarget + "-renewXNumberOfDaysBeforeExpiration"] = $renewXNumberOfDaysBeforeExpiration
+$appSettingsTarget["letsencrypt:" + $webAppTarget + "-resourceGroup"] = $resourceGroupTarget
+$appSettingsTarget["letsencrypt:" + $webAppTarget + "-subscriptionId"] = $subscriptionId
+$appSettingsTarget["letsencrypt:" + $webAppTarget + "-tenantId"] = $tenantId 
+if ($appSettingsTarget.ContainsKey("letsencrypt:webApps")) {
+	if (!$appSettingsTarget["letsencrypt:webApps"].ToLower().Contains($webAppTarget.ToLower())) {
```
Comment (ohadschn) [inline WebAppConfig.ps1]: It is a better practice to use `ToUpperInvariant`, or even better `CompareOptions.IgnoreCase`: https://stackoverflow.com/a/15464440/67824

File: WebAppConfig.ps1 [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+# Create Hash variable for App Settings
+$appSettingsTarget = @{}
+
+# Copy over all Existing App Settings to the Hash
+ForEach ($appSettingSource in $appSettingsSource) {
+    $appSettingsTarget[$appSettingSource.Name] = $appSettingSource.Value
+}
+
+# Add new settings
+$appSettingsTarget["letsencrypt:" + $webAppTarget + "-clientId"] = $clientId
```
Comment (ohadschn) [inline WebAppConfig.ps1]: Extract `letsencrypt:` to variable/constant

File: WebAppConfig.ps1 (hunk 1 above)
Comment (chrfin) [reply WebAppConfig.ps1]: I will add these settings, thanks for the tip. I'm not very experienced with PS and did not know of them...

File: WebAppConfig.ps1 (hunk 2 above)
Comment (chrfin) [reply WebAppConfig.ps1]: I do not (completele) agree here. Yes, these values should not be checked in, but I download the script, enter those values and save the modified script as "WebAppConfig_NAME.ps1". Then I can re-execute this script each time I add an app to NAME. I maintain several AppServices in different directories so this is the most comfortable way for me. Of course you do not need to adapt/set this default values and can always provide them via a parameter, but I do not want to force that...

File: WebAppConfig.ps1 (hunk 3 above)
Comment (chrfin) [reply WebAppConfig.ps1]: Will update that...

File: WebAppConfig.ps1 (hunk 4 above)
Comment (chrfin) [reply WebAppConfig.ps1]: Good idea!

File: WebAppConfig.ps1 (hunk 5 above)
Comment (chrfin) [reply WebAppConfig.ps1]: True...

File: WebAppConfig.ps1 (hunk 6 above)
Comment (chrfin) [reply WebAppConfig.ps1]: Good idea!

**Pull request merged**